From: Yifeng <wuyifeng@nd.com.cn>
Date: Sat, 16 Jul 2022 07:40:00 +0000
Subject: Disable auto updater and welcome screen

These two features are added by bromise and we don't need them, so disable them.

---
 chrome/android/java/src/org/chromium/chrome/browser/about_settings/AboutChromeSettings.java | 1 +
 chrome/android/java/src/org/chromium/chrome/browser/omaha/inline/BromiteInlineUpdateController.java | 2 --
 chrome/browser/first_run/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunStatus.java | 4 ++--
 3 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/about_settings/AboutChromeSettings.java b/chrome/android/java/src/org/chromium/chrome/browser/about_settings/AboutChromeSettings.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/about_settings/AboutChromeSettings.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/about_settings/AboutChromeSettings.java
@@ -65,9 +65,10 @@ public class AboutChromeSettings
         p = findPreference(PREF_LEGAL_INFORMATION);
         int currentYear = Calendar.getInstance().get(Calendar.YEAR);
         p.setSummary(getString(R.string.legal_information_summary, currentYear));

         ChromeSwitchPreference allowInlineUpdate =
                 (ChromeSwitchPreference) findPreference(PREF_ALLOW_INLINE_UPDATE);
+        allowInlineUpdate.setVisible(false);
         allowInlineUpdate.setChecked(
             OmahaBase.getSharedPreferences()
                     .getBoolean(OmahaBase.PREF_ALLOW_INLINE_UPDATE, false));
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/omaha/inline/BromiteInlineUpdateController.java b/chrome/android/java/src/org/chromium/chrome/browser/omaha/inline/BromiteInlineUpdateController.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/omaha/inline/BromiteInlineUpdateController.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/omaha/inline/BromiteInlineUpdateController.java
@@ -84,11 +84,9 @@ public class BromiteInlineUpdateController implements InlineUpdateController {

         mEnabled = enabled;
         // check for an update when state changes
-        if (mEnabled) pullCurrentState();
     }

     @Override
     public @Nullable @UpdateStatusProvider.UpdateState Integer getStatus() {
-        if (mEnabled) pullCurrentState();
         return mUpdateState;
     }
diff --git a/chrome/browser/first_run/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunStatus.java b/chrome/browser/first_run/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunStatus.java
--- a/chrome/browser/first_run/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunStatus.java
+++ b/chrome/browser/first_run/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunStatus.java
@@ -44,7 +44,7 @@ public class FirstRunStatus {
      */
     public static boolean getFirstRunFlowComplete() {
         return SharedPreferencesManager.getInstance().readBoolean(
-                ChromePreferenceKeys.FIRST_RUN_FLOW_COMPLETE, false);
+                ChromePreferenceKeys.FIRST_RUN_FLOW_COMPLETE, true);
     }

     /**
@@ -62,7 +62,7 @@ public class FirstRunStatus {
      */
     public static boolean shouldSkipWelcomePage() {
         return SharedPreferencesManager.getInstance().readBoolean(
-                ChromePreferenceKeys.FIRST_RUN_SKIP_WELCOME_PAGE, false);
+                ChromePreferenceKeys.FIRST_RUN_SKIP_WELCOME_PAGE, true);
     }

     /**
--